### PR TITLE
Build new Ubuntu 18.04 LTS OCI page

### PR DESCRIPTION
**Pull request by carkod**
## Done

- Built as per [doc](https://docs.google.com/document/d/1Kdq2tmLk4aA6wceyhS9gmmmyfCU8Ralw0MOts6HtmOM)
- Design is meant to match the rest of the pages in the 18-04 bubble, please compare against those

## QA

- Check https://ubuntu-com-12917.demos.haus/18-04/oci against [copy doc](https://docs.google.com/document/d/1Kdq2tmLk4aA6wceyhS9gmmmyfCU8Ralw0MOts6HtmOM)
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
    - Be sure to test on mobile, tablet and desktop screen sizes
- [List additional steps to QA the new features or prove the bug has been resolved]

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-4152

### CODE REVIEW

**Comment by juanruitina** [comment]
Demo doesn't work!

**Comment by nottrobin** [comment]
The demo log says:

    error: deployment "ubuntu-com-12917-demos-haus" exceeded its progress deadline

Nine times out of ten, what this means is that the application has failed to start because it has an error.

It looks like that's the case here, here is the log from the container in question:

<details>
<summary>Log from <code>kubectl logs pod/ubuntu-com-12917-demos-haus-7d6cc9b85b-rs5gl</code></summary>
<pre><code>
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/srv/webapp/app.py", line 35, in <module>
    from webapp.handlers import init_handlers
  File "/srv/webapp/handlers.py", line 6, in <module>
    from webapp.context import (
  File "/srv/webapp/context.py", line 19, in <module>
    api_session = CachedSession(fallback_cache_duration=3600)
  File "/root/.local/lib/python3.10/site-packages/canonicalwebteam/http/__init__.py", line 153, in __init__
    cache = FileCache(file_cache_directory)
  File "/root/.local/lib/python3.10/site-packages/cachecontrol/caches/file_cache.py", line 84, in __init__
    raise ImportError(notice)
ImportError: 
NOTE: In order to use the FileCache you must have
filelock installed. You can install it via pip:
  pip install filelock

2023-06-05 08:43:42.550Z ERROR gunicorn.error "Exception in worker process"
Traceback (most recent call last):
  File "/root/.local/lib/python3.10/site-packages/cachecontrol/caches/file_cache.py", line 73, in __init__
    from filelock import FileLock
ModuleNotFoundError: No module named 'filelock'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/root/.local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 589, in spawn_worker
    worker.init_process()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/workers/ggevent.py", line 146, in init_process
    super().init_process()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/root/.local/lib/python3.10/site-packages/talisker/gunicorn.py", line 156, in load_wsgiapp
    app = super(TaliskerApplication, self).load_wsgiapp()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/root/.local/lib/python3.10/site-packages/gunicorn/util.py", line 359, in import_app
    mod = importlib.import_module(module)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/srv/webapp/app.py", line 35, in <module>
    from webapp.handlers import init_handlers
  File "/srv/webapp/handlers.py", line 6, in <module>
    from webapp.context import (
  File "/srv/webapp/context.py", line 19, in <module>
    api_session = CachedSession(fallback_cache_duration=3600)
  File "/root/.local/lib/python3.10/site-packages/canonicalwebteam/http/__init__.py", line 153, in __init__
    cache = FileCache(file_cache_directory)
  File "/root/.local/lib/python3.10/site-packages/cachecontrol/caches/file_cache.py", line 84, in __init__
    raise ImportError(notice)
ImportError: 
NOTE: In order to use the FileCache you must have
filelock installed. You can install it via pip:
  pip install filelock

2023-06-05 08:43:42.705Z WARNING talisker.context "track API called when there is no active context, data will not be stored" type=http duration=161.371
2023-06-05 08:43:42.706Z INFO talisker.requests "http request" url=https://sentry.is.canonical.com/api/13/store/ method=POST host=sentry.is.canonical.com status_code=200 server="Apache/2.4.29 (Ubuntu)" duration_ms=161.371 request_type=application/octet-stream request_size=3072 response_type=application/json response_size=41
2023-06-05 08:43:42.707Z INFO gunicorn.error "Worker exiting (pid: 102)"
2023-06-05 08:43:42.771Z WARNING talisker.context "track API called when there is no active context, data will not be stored" type=http duration=208.56
2023-06-05 08:43:42.772Z INFO talisker.requests "http request" url=https://sentry.is.canonical.com/api/13/store/ method=POST host=sentry.is.canonical.com status_code=200 server="Apache/2.4.29 (Ubuntu)" duration_ms=208.56 request_type=application/octet-stream request_size=3075 response_type=application/json response_size=41
2023-06-05 08:43:42.773Z INFO gunicorn.error "Worker exiting (pid: 103)"
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgreenlet.Greenlet.run
  File "/root/.local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
    self.reap_workers()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
    raise HaltServer(reason, self.WORKER_BOOT_ERROR)
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
2023-06-05T08:43:42Z <Greenlet at 0x7f4362ffd900: <bound method Arbiter.handle_chld of <gunicorn.arbiter.Arbiter object at 0x7f43627c27a0>>(17, None)> failed with HaltServer

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgreenlet.Greenlet.run
  File "/root/.local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 242, in handle_chld
    self.reap_workers()
  File "/root/.local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
    raise HaltServer(reason, self.WORKER_BOOT_ERROR)
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
2023-06-05T08:43:42Z <Greenlet at 0x7f4362ffd900: <bound method Arbiter.handle_chld of <gunicorn.arbiter.Arbiter object at 0x7f43627c27a0>>(17, None)> failed with HaltServer
</code></pre>
</details>

Looks like maybe you introduced a dependency on "filelock"?

**Comment by nottrobin** [comment]
@carkod you're right, this was subtly introduced by a background upgrade to cachecontrol I think. This wouldn't have happened if we were using poetry :)

I've filed https://github.com/canonical/ubuntu.com/pull/12924

**Comment by juanruitina** [comment]
Can you align the code for webinars with [MariaPaula's](https://github.com/canonical/ubuntu.com/pull/12915)? Also, please replace the thumbnail for the first one with [this](https://assets.ubuntu.com/v1/a8ecc110-18-04_lts_eos_webinar.png). Thank you!

**Comment by juanruitina** [comment]
There is still an empty paragraph below the first thumbnail, but other than that, LGTM.

**Comment by julpal7** [comment]
Hi @carkod, thanks for aligning it with the rest of the pages. I have noticed some issues:

1. Please replace the screenshots with the updated assets here (to keep the outline style consistent):
[LTS end of support](https://assets.ubuntu.com/v1/f52739bc-18.04_LTS_End_of_Standard_Support.png)
[LTS migrating to 20.04](https://assets.ubuntu.com/v1/2aceaa9c-Migrating_to_20.04_LTS.png)

2. This section has responsiveness issues. It doesn't scale properly between 620px and 1035px width.
Screenshot 2023-06-06 at 10 51 12

3. I have a question about this button. The copy doc does not indicate the link should be boxed, I would interpret it as analogical to the "Get in touch" link in the next section. However, then we would be left completely without a CTA at this page (on the other 18.04 ones we have green CTAs throughout). @juanruitina do you have any thoughts?

Screenshot 2023-06-06 at 10 07 20
<br>Reference from the copy doc:
Screenshot 2023-06-06 at 10 21 01

**Review by mtruj013 — CHANGES_REQUESTED** [review]
Thanks @carkod! A few comments in the code and could you also fix the cols for medium screens? I believe it's supposed to be a 50/50 split for the most part

**Comment by carkod** [comment]
> Hi @carkod, thanks for aligning it with the rest of the pages. I have noticed some issues:
> 
> 1. Please replace the screenshots with the updated assets here (to keep the outline style consistent):
>    [LTS end of support](https://assets.ubuntu.com/v1/f52739bc-18.04_LTS_End_of_Standard_Support.png)
>    [LTS migrating to 20.04](https://assets.ubuntu.com/v1/2aceaa9c-Migrating_to_20.04_LTS.png)
> 2. This section has responsiveness issues. It doesn't scale properly between 620px and 1035px width.
> 
> 3. I have a question about this button. The copy doc does not indicate the link should be boxed, I would interpret it as analogical to the "Get in touch" link in the next section. However, then we would be left completely without a CTA at this page (on the other 18.04 ones we have green CTAs throughout). @juanruitina do you have any thoughts?


Hi @julpal7:

1. The thumbnails you provided are exactly the same as the ones I have there?


> Hi @carkod, thanks for aligning it with the rest of the pages. I have noticed some issues:
> 
> 1. Please replace the screenshots with the updated assets here (to keep the outline style consistent):
>    [LTS end of support](https://assets.ubuntu.com/v1/f52739bc-18.04_LTS_End_of_Standard_Support.png)
>    [LTS migrating to 20.04](https://assets.ubuntu.com/v1/2aceaa9c-Migrating_to_20.04_LTS.png)
> 2. This section has responsiveness issues. It doesn't scale properly between 620px and 1035px width.
> 
> 3. I have a question about this button. The copy doc does not indicate the link should be boxed, I would interpret it as analogical to the "Get in touch" link in the next section. However, then we would be left completely without a CTA at this page (on the other 18.04 ones we have green CTAs throughout). @juanruitina do you have any thoughts?
> 

We have plenty of pages without CTAs, I think they are used to emphasize a specific action, I don't think they are required to be present just for the sake consistency.

File: templates/18-04/oci.html [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  <div class="row">
+    <div class="col-6">
+      <h2>
+        Security updates for <br class="u-hide--small u-hide--medium" />Ubuntu 18.04 LTS, with Ubuntu Pro
+      </h2>
+    </div>
+    <div class="col-6">
+      <div class="p-strip is-shallow u-no-padding--top">
+        <p>
+          The Ubuntu Security Team is dedicated to providing timely security updates for all packages on Ubuntu 18.04 LTS for <i>x86-64</i> and <i>arm64</i> architectures.
```
[the remark this answers is not in the document]
Comment (carkod) [reply templates/18-04/oci.html]: ok, removed. It was italic in the copy doc

File: templates/18-04/oci.html [hunk 2]
```diff
@@ -0,0 +1,311 @@
+{% extends "18-04/base_18-04.html" %}
+
+{% block title %}Ubuntu 18.04 LTS (Bionic Beaver){% endblock %}
+{% block meta_description %}Ubuntu 18.04 LTS (Bionic Beaver) is out of standard support on 31 May 2023. Ubuntu Pro provides continued support and security coverage for Ubuntu 18.04 LTS for additional 5 years, until 2028.{% endblock %}
```
Comment (mtruj013) [inline templates/18-04/oci.html:4]:
```suggestion
{% block meta_description %}Ubuntu 18.04 LTS (Bionic Beaver) is out of standard support on 31 May 2023. Ubuntu Pro provides continued support and security coverage for Ubuntu 18.04 LTS for an additional 5 years, until 2028.{% endblock %}
```
Typo on the doc

**Review by mtruj013 — CHANGES_REQUESTED** [review]
Thanks for the changes @carkod, just a few more suggestions. There are still quite a few sections that are not aligned correctly for medium screens, could you please cross reference against either of the new pages for this bubble? Regarding the CTA I think we need to wait for @juanruitina's input

File: templates/18-04/oci.html [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        <p>
+          Standard Security Maintenance of an Ubuntu LTS release covers binary packages that reside in the <a href="/community/Repositories/Ubuntu">Main Ubuntu repository</a> for a period of five years. For continued security beyond the standard five-year maintenance period, Ubuntu Pro delivers security maintenance to a wide range of binary packages that are commonly used in cloud and server workloads on 64-bit x86 AMD/Intel architectures for a period of five years beyond the end of standard support.
+        </p>
+        <p>
+          Additionally, Ubuntu Pro can also cover all packages in the <a href="https://help.ubuntu.com/community/Repositories/Ubuntu">Ubuntu Universe repository</a>, until April 2028.
+        </p>
+      </div>
+    </div>
+  </div>
+  <div class="row">
```
Comment (mtruj013) [inline templates/18-04/oci.html:181]: Thanks for fixing the medium screen cols! This section still looks wonky on large screens though, could you take another look?
Screenshot from 2023-06-07 10-30-50

Comment (carkod) [reply templates/18-04/oci.html:181]: Fixed. Thanks

**Review by mtruj013 — APPROVED** [review]
Thanks @carkod, LGTM!

**Review by mtruj013 — APPROVED** [review]
Thanks @carkod, LGTM!

**Pull request merged**